pkg/agent/billing/billing.go

```diff
@@ -18,14 +19,26 @@ import (
 )
 
 type Config struct {
-	URL                       string `json:"url"`
-	CPUMetricName             string `json:"cpuMetricName"`
-	ActiveTimeMetricName      string `json:"activeTimeMetricName"`
-	CollectEverySeconds       uint   `json:"collectEverySeconds"`
-	AccumulateEverySeconds    uint   `json:"accumulateEverySeconds"`
-	PushEverySeconds          uint   `json:"pushEverySeconds"`
-	PushRequestTimeoutSeconds uint   `json:"pushRequestTimeoutSeconds"`
-	MaxBatchSize              uint   `json:"maxBatchSize"`
+	Clients                ClientsConfig `json:"clients"`
+	CPUMetricName          string        `json:"cpuMetricName"`
+	ActiveTimeMetricName   string        `json:"activeTimeMetricName"`
+	CollectEverySeconds    uint          `json:"collectEverySeconds"`
+	AccumulateEverySeconds uint          `json:"accumulateEverySeconds"`
+}
+
+type ClientsConfig struct {
```

Comment:
I'd use a single `ClientConfig` with a type field. If we will need more fields for the S3, that's fine, those fields will be empty for HTTP. This will allow to have multiple clients of the same type.

But the current implementation is also fine.

Reply:
Yeah, I wanted to do that, but felt like dealing with the JSON unmarshaling and validation was going to be more annoying that way.